test.js

```diff
@@ -1581,6 +1581,247 @@ test('errors', function (t) {
+      url: '/sign',
+      payload: { foo: 'bar' }
+    }).then(function (signResponse) {
+      const token = JSON.parse(signResponse.payload).token
+      t.ok(token)
+
+      fastify.inject({
+        method: 'get',
+        url: '/verify',
+        cookies: {
+          jwt: ''
+        },
+        headers: {
+          authorization: `Bearer ${token}`
+        }
+      }).then(function (verifyResponse) {
+        const decodedToken = JSON.parse(verifyResponse.payload)
+        t.is(decodedToken.foo, 'bar')
+      })
+    })
+  })
+
+  t.test('both authorization and cookie headers present, both values empty', function (t) {
+    t.plan(3)
+    fastify.inject({
+      method: 'post',
+      url: '/sign',
+      payload: { foo: 'bar' }
+    }).then(function (signResponse) {
+      const token = JSON.parse(signResponse.payload).token
+      t.ok(token)
+
+      fastify.inject({
+        method: 'get',
+        url: '/verify',
+        cookies: {
+          jwt: ''
+        },
+        headers: {
+          authorization: ''
+        }
+      }).then(function (verifyResponse) {
+        const error = JSON.parse(verifyResponse.payload)
+        t.is(error.message, 'No Authorization was found in request.cookies')
+        t.is(error.statusCode, 401)
+      })
+    })
+  })
+
+  t.test('both authorization and cookie headers present, header malformed', function (t) {
+    t.plan(3)
+    fastify.inject({
+      method: 'post',
+      url: '/sign',
+      payload: { foo: 'bar' }
+    }).then(function (signResponse) {
+      const token = JSON.parse(signResponse.payload).token
+      t.ok(token)
+
+      fastify.inject({
+        method: 'get',
+        url: '/verify',
+        cookies: {
+          jwt: token
+        },
+        headers: {
+          authorization: 'BearerX'
+        }
+      }).then(function (verifyResponse) {
+        const error = JSON.parse(verifyResponse.payload)
+        t.is(error.message, 'Format is Authorization: Bearer [token]')
+        t.is(error.statusCode, 400)
+      })
+    })
+  })
+})
+
+test('token in cookie, without fastify-cookie parsing', function (t) {
+  t.plan(2)
+
+  const fastify = Fastify()
+  fastify.register(jwt, { secret: 'test', cookie: { cookieName: 'jwt' } })
+
+  fastify.post('/sign', function (request, reply) {
+    return reply.jwtSign(request.body)
+      .then(function (token) {
+        return { token }
+      })
+  })
+
+  fastify.get('/verify', function (request, reply) {
+    return request.jwtVerify()
+      .then(function (decodedToken) {
+        return reply.send(decodedToken)
+      })
+      .catch(function (error) {
+        return reply.send(error)
+      })
+  })
+
+  t.test('token present in cookie, but unparsed', function (t) {
+    t.plan(3)
+    fastify.inject({
+      method: 'post',
+      url: '/sign',
+      payload: { foo: 'bar' }
+    }).then(function (signResponse) {
+      const token = JSON.parse(signResponse.payload).token
+      t.ok(token)
+
+      fastify.inject({
```

Comment:
I think you should be able to return this and avoid nesting of promises.

Reply:
I have gone ahead and added `return` to all nested promise chain cases.

Reply:
Though I am not sure if the same should be done to the parent. But I will assume its not absolutely necessary.